src/ssl.c

```diff
@@ -28736,6 +28736,9 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
                 return NULL;
             if (XFSEEK(bp->file, i, SEEK_SET) != 0)
                 return NULL;
+            /* check calculated length */
```

Comment:
It looks like this check should also be done for the WOLFSSL_BIO_MEMORY case above and having it down below made sure it was checked for all types. Is there a better way to fix this warning?

Reply:
Can you check into moving this check back down so its done for all cases (future proof it).

Reply:
This was the cause of ZD4044. From the discussion there:
> With the NO_FILESYSTEM define, i is only ever assigned the zero before the if (l - i < 0) statement. It's not modified anywhere after the initiazlization.
>
> And l is checked to be larger than zero: if (l <= 0) return NULL; 
> After that it is not modified until the if (l - i < 0) statement.

I am fine with marking it as "will not fix" because it does seem like a valid check to prevent problems with new code.

Reply:
I think I have better solution:
```
@@ -28796,9 +28799,12 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
                 return NULL;
             if (XFSEEK(bp->file, i, SEEK_SET) != 0)
                 return NULL;
+
             /* check calculated length */
             if (l - i < 0)
                 return NULL;
+
+            l -= i;
 #else
             WOLFSSL_MSG("Unable to read file with NO_FILESYSTEM defined");
             return NULL;
@@ -28807,7 +28813,7 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
         else
             return NULL;
 
-        pem = (unsigned char*)XMALLOC(l - i, 0, DYNAMIC_TYPE_PEM);
+        pem = (unsigned char*)XMALLOC(l, 0, DYNAMIC_TYPE_PEM);
         if (pem == NULL)
             return NULL;
```